generate_report_excels.py

```
@@ -0,0 +1,782 @@
+                        'e.g: 2024-01-01T08:30:00Z to filter reports created before this date. Only allowed in manual mode.')
+    parser.add_argument('--created_after', type=str, default=None,
+                        help='The date string in the format YYYY-MM-DD\'T\'HH:MM:SS\'Z\''
+                        'e.g: 2024-01-01T08:30:00Z to filter reports created after this date. Only allowed in manual mode.')
+    args = parser.parse_args()
+    # Validate inputs
+    if args.created_before == "" or args.created_after == "":
+        logging.error(
+            "Invalid date format for created_before or created_after.")
+        raise SystemExit
+    if args.created_before and args.created_after:
+        if args.created_before <= args.created_after:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)